diff --git a/node_modules/@secretarium/react-native-ble/ios/C3POReactNativeBle.swift b/node_modules/@secretarium/react-native-ble/ios/C3POReactNativeBle.swift
index 3c19ffa..28aea21 100644
--- a/node_modules/@secretarium/react-native-ble/ios/C3POReactNativeBle.swift
+++ b/node_modules/@secretarium/react-native-ble/ios/C3POReactNativeBle.swift
@@ -14,6 +14,7 @@ open class C3POReactNativeBle: RCTEventEmitter, CBPeripheralManagerDelegate, CBC
 
     var manufacturerId: Int?;
     var peripheralUUID: CBUUID?;
+    var peripheralName: String = "iOS BLE Device"
     var peripheralManager: CBPeripheralManager?
     var peripheralReady: Bool = false
     var peripheralShouldLive: Bool = false
@@ -82,7 +83,7 @@ open class C3POReactNativeBle: RCTEventEmitter, CBPeripheralManagerDelegate, CBC
 
         self.peripheralManager?.removeAllServices()
         self.peripheralManager?.add(CBMutableService(type: self.peripheralUUID!, primary: true))
-        self.peripheralManager?.startAdvertising([CBAdvertisementDataLocalNameKey : "C3PO", CBAdvertisementDataServiceUUIDsKey : [self.peripheralUUID!]])
+        self.peripheralManager?.startAdvertising([CBAdvertisementDataLocalNameKey : "worldlines-" + self.peripheralName, CBAdvertisementDataServiceUUIDsKey : [self.peripheralUUID!]])
     }
 
     @objc
@@ -91,9 +92,10 @@ open class C3POReactNativeBle: RCTEventEmitter, CBPeripheralManagerDelegate, CBC
     }
 
     @objc
-    func broadcast(_ uuid: NSString, payload: NSArray, resolver resolve : RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+    func broadcast(_ uuid: NSString, name: String, payload: NSArray, resolver resolve : RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
         self.peripheralUUID = CBUUID(string: uuid as String)
         self.peripheralShouldLive = true
+        self.peripheralName = name
         startBroadcast()        
         resolve(true)
     }
diff --git a/node_modules/@secretarium/react-native-ble/src/index.tsx b/node_modules/@secretarium/react-native-ble/src/index.tsx
index 20903d0..d03a6a1 100644
--- a/node_modules/@secretarium/react-native-ble/src/index.tsx
+++ b/node_modules/@secretarium/react-native-ble/src/index.tsx
@@ -2,7 +2,7 @@ import { NativeModules } from 'react-native';
 
 type C3POReactNativeBleType = {
     setManufacturerId(manufacturerId: number): void;
-    broadcast(uuid: string, payload: number[]): Promise<boolean | Error>;
+    broadcast(uuid: string, name: string, payload: number[]): Promise<boolean | Error>;
     stopBroadcast(): Promise<boolean>;
     scan(): Promise<boolean>;
     stopScan(): Promise<boolean>;
